TVShows/TVShows/Modules/ShowDetails/ShowDetailsInteractor.swift

```diff
@@ -0,0 +1,63 @@
+//
+//  ShowDetailsInteractor.swift
+//  TVShows
+//
+//  Created by Ivan Milicevic on 22/10/2018.
+//  Copyright (c) 2018 Ivan Milicevic. All rights reserved.
+//
+//  This file was generated by the 🐍 VIPER generator
+//
+
+import UIKit
+import Alamofire
+import CodableAlamofire
+
+final class ShowDetailsInteractor {
+}
+
+// MARK: - Extensions -
+
+extension ShowDetailsInteractor: ShowDetailsInteractorInterface {
+
+    func fetchShowDetails(show: Show, completion: @escaping (Result<ShowDetails>) -> ()) {
+        guard
+            let token = NetworkManager.loginData?.token
+            else {
+                return
```

Comment:
And maybe you could call completion with handcrafted `Result` with your custom error to notify user that his session has expired.

Also, take a look at Alamofire's `SessionManager`  and `RequestAdapter` to avoid constantly adding authorization header in every request ;)